File: Python/ex4_calculator.py
# 4. Create simple calculator

def digit():
    number = input("Type a digit: ")
    while not number.isdigit():
        number = input("Wrong digit, please type a correct digit: ")
    return int(number)

if __name__ == "__main__":
    try:
        outcome = digit()
        while True:
            choice = str(input("Type an operator (+, -, *, /, =): "))
            match choice.strip():
                case "+":  
                    outcome += digit()
                case "-":
                    outcome -= digit()
                case "*":
                    outcome *= digit()
                case "/":
                    outcome /= digit()
                case "=":
                    print("Outcome =", outcome)
                    break
                case _:
                    print("Wrong operator")
    except Exception as wrong_user:
        print(f"Error!!, program shut down due to {wrong_user}")

File: JavaScript/ex4_calculator.js
// 4. Create simple calculator

const prompt = require ("prompt-sync")();

function digit(){
    let number = prompt("Type a digit: ");
    while (Number.isInteger(+number) !== true){
        number = prompt("Wrong digit, please type a correct digit: ");
    }
    return parseInt(number, 10);
}

if (require.main === module){
    try {
        let outcome = digit();
        while (true){
            let choice = String(prompt("Type an operator (+, -, *, /, =): "));
            switch (choice.trim()){
                case ("+"):
                    outcome += digit();
                    break;
                case ("-"):
                    outcome -= digit();
                    break;
                case ("*"):
                    outcome *= digit();
                    break;
                case ("/"):
                    outcome /= digit();
                    break;
                case ("="):
                    console.log("Outcome =", outcome)
                    return outcome;
                default:
                    console.log("Wrong operator")
                    break;
            }
        }
    }
    catch (error) {
        console.error(error)
    }
}